src/App.tsx

```diff
@@ -14,11 +14,106 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [visibleList, setVisibleList] = useState(false);
 
-export default App;
+  const isVisibleList = () => {
+    setVisibleList(!visibleList);
+  };
+
+  const makeReverse = () => {
+    setGoods([...goods].reverse());
+  };
+
```

Comment:
```suggestion
const makeReverse = () => {
    setGoods([...goods].reverse());
  };
    const makeSortAlphabet = () => {
    setGoods([...goods].sort((a, b) => a.localeCompare(b)));
  };

  const makeSortLength = () => {
    setGoods([...goods].sort((a, b) => a.length - b.length));
  };
```
If you make some changes depending on the value of previous state, you should use functional setState:
 setGoods((currentGoods) => ({...}))